File: core/src/main/java/com/canyon/logger/LoggerInjectProvider.java
package com.canyon.logger;

import com.canyon.inject.InjectProvider;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class LoggerInjectProvider implements InjectProvider {
    @Override
    public boolean isMatch(Class<?> clazz) {
        return clazz.equals(Logger.class);
    }

    @Override
    public <T> T create(Class<T> objClazz, Class<T> fieldClazz) {
        return (T) LoggerFactory.getLogger(objClazz);
    }
}

File: core/src/main/kotlin/com/canyon/logger/LoggerInjectProvider.kt
package com.canyon.logger

import com.canyon.inject.InjectProvider
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import kotlin.reflect.KClass
import kotlin.reflect.jvm.jvmName

class LoggerInjectProvider : InjectProvider {
    override fun isMatch(kClass: KClass<*>): Boolean {
        return kClass == Logger::class
    }
    override fun create(objClass: KClass<*>, fieldClass: KClass<*>): Any {
        return LogManager.getLogger(objClass.jvmName)
    }
}